pkg/core/pipeline/resource/main.go

```diff
@@ -60,6 +61,9 @@ func New(rs ResourceConfig) (resource Resource, err error) {
 	switch kind {
 	case "aws/ami":
 		return awsami.New(rs.Spec)
+	case "cargopackage":
+
+		return cargopackage.New(rs.Spec, rs.SCMID)
```

Comment:
@olblak I don't know if you are ok with using a `New` signature that is different than the other resource definition? By having `scmid` here, I can infer the proper index to use in the `source` method without having to rely on comparing the `workingdir` to the current dir

Reply:
Changing this new signature is fine that being said I don't understand why there is a different behavior if it's a scm or not.

In the case of the source 
```func (cp CargoPackage) Source(workingDir string) (string, error) {```
workingDir is already correctly configured either to the scm directory or local path and it should only be used fi `cp.spec.IndexDir` and `cp.spec.IndexURL` is not set

Reply:
If we are in the case of using an `scm` then the `Source` function get the scm directory as input, `cp.spec.IndexDir` and `cp.spec.IndexURL` will not be set and we should use the `workingDir`.

If we are in the case of **not** using an `scm` but we want to use the default cargo index, then we will get the local path as input and `cp.spec.IndexURL` will not be set . 

The only way to differentiate the two scenario from within the `Source` function is then to compare that the `workingDir` is not the `localPath` and it seemed more hacky than to give the `scm` information in the `New` function

Reply:
Thanks for the information, then I guess a boolean `isSCM` could make sense in the New signature

Reply:
So you would pefere `rs.SCMID != ""` than passing the scmId?

Reply:
done